extensions/azure/cosmos/contract-definition-store-cosmos/src/test/java/org/eclipse/dataspaceconnector/contract/definition/store/CosmosContractDefinitionStoreTest.java

```diff
@@ -127,12 +127,22 @@ void update() {
     }
 
     @Test
-    void delete() {
-        store.delete("some-id");
-
+    void deleteById_whenMissing_returnsNull() {
+        var contractDefinition = store.deleteById("some-id");
+        assertThat(contractDefinition).isNull();
         verify(cosmosDbApiMock).deleteItem(notNull());
     }
 
+    @Test
+    void delete_whenContractDefinitionPresent_deletes() {
```

Comment:
How about adding one more test where we validate the behavior when EdcPersistenceException is thrown by cosmosDbApiMock ?

Reply:
Yes it makes sense thanks! I added a test called delete_whenCosmoDbApiThrows_throws.